Plugins/Json/src/JsonDetectorElement.cpp

```diff
@@ -0,0 +1,40 @@
+// This file is part of the ACTS project.
+//
+// Copyright (C) 2016 CERN for the benefit of the ACTS project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "Acts/Plugins/Json/JsonDetectorElement.hpp"
+
+#include "Acts/Plugins/Json/SurfaceJsonConverter.hpp"
+
+namespace Acts {
+
+JsonDetectorElement::JsonDetectorElement(const nlohmann::json &jSurface,
+                                         double thickness)
+    : m_thickness(thickness) {
+  m_surface = Acts::SurfaceJsonConverter::fromJson(jSurface);
+  m_transform = m_surface->transform({});
```

Comment:
Why do we need the stored `m_transform` here actually?

Reply:
> Can't we read the transform from `jSurface` directly, instead of having the geometry context be default-constructed?

Is this not exactly what the  surface construction + default constructed geometry context does?

Reply:
I mean that we can extract the transform from the json with a default constructed geometry context, and then we link the surface to this detector element, so afterwards the geometry context is used as intended.